_posts/2017-09-12-repeatable-serverless-applications.markdown

```diff
@@ -0,0 +1,88 @@
+---
+layout: post
+title: "Repeatable Serverless Applications"
+date: 2017-09-11
+categories: serverless architecture docker aws
+author: Rick Foxcroft
+image: img/no_servers.png
+---
+
+Servers can be expensive right? We aren't the biggest fan of lengthy provisioning processes, managing, maintenance and the costly nature that comes with deploying applications on servers. This doesn't mean that we avoid them altogether. We still have a monolithic application that uses several ec2 instances for numerous environments. On one of our recent mob-programming sessions, we decided to tackle the topic of, "How can we turn off any lurking, superfluous and idle ec2 instances" or any other idle instances we may have forgotten to stop/terminate. 
+
+![serverless benefits]({{ site.url }}/img/about_serverless_architecture.jpg)
+
+The preferred approach for any new software project here at Mechanical Rock, is to tackle it with a test-first mindset and a serverless implementation. The difficulty comes in, with the infancy of the whole "serverless" paradigm. Meaning, that there aren't many established best practices regarding test-first serverless applications and as a DevOps consultancy, ideally we would like to practice a Continuous Delivery model on all of our software projects. In order to achieve this, we need to be able test our application locally, on a pipeline and have it run bug-free in the cloud. The good thing is, there are tools out there, that can help us achieve this, most desired method of software development efficiency. 
+
+![serverless benefits]({{ site.url }}/img/localstack.png)
+
+Meet, LocalStack. LocalStack, was created by the good people at Atlassian and provides you with "A fully functional local AWS cloud stack". This means, that with a few cheeky bash scripts and some Docker magic. You can automate the provisioning of all(almost) the necessary resources for your cloud-based application. Sounds too easy? Well you're absolutely right! With a docker-compose file, you can start up the LocalStack service in conjunction with your dev-env in your Dockerfile. 
```

Comment:
Sentence is broken "with a few cheeky bash scripts and some Docker magic. You can automate the..." should be a comma in the middle?